### DISCUSSION TITLE
Issues during upgrade to 2.11.x

### DISCUSSION THREAD

**Discussion by tdrevvatne** [netbox-community/netbox#6231] [Q&A]
Having an instance up to date at 2.10.10, upgrading have been straight forward up till now..
Trying to go further up to 2.11.x, there are issue..

These messages will probably make sence to lots of you, but some (like me) struggle..:

  Applying ipam.0044_standardize_models...Traceback (most recent call last):
  File "/opt/netbox-2.11.1/venv/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.InsufficientPrivilege: must be owner of relation ipam_vlan_id_seq

Thanks.

**Comment by candlerb** [comment]
Can you get a postgres shell (typically `sudo -u postgres psql netbox`),then do the following and show the results - I'm showing what I see on my system here.

```
netbox=# \dt ipam_vlan
          List of relations
 Schema |   Name    | Type  | Owner
--------+-----------+-------+--------
 public | ipam_vlan | table | netbox
(1 row)

netbox=# \ds ipam_vlan_id_seq
               List of relations
 Schema |       Name       |   Type   | Owner
--------+------------------+----------+--------
 public | ipam_vlan_id_seq | sequence | netbox
(1 row)
```

**Reply by candlerb** [reply]
Based on a similar problem at #6233:

1. Try running this command (it only reads from the database and spits out some SQL to stdout)

    ```
    sudo -u postgres pg_dump -s netbox | grep '^ALTER.*OWNER TO'
    ```

    Do you see a mixture of owners after `OWNER TO`?

2. If so, change the command as follows (again, at this point it only reads from the database)

    ```
    sudo -u postgres pg_dump -s netbox | grep '^ALTER.*OWNER TO' | sed 's/OWNER TO .*;/OWNER TO netbox;/'
    ```

3. If the output from (2) looks reasonable, then pipe it back into psql to update your database.  Make sure you have a backup first.

**Comment by jeremystretch** [comment]
I strongly suggest you investigate what changes were made to your database, because there's nothing about the v2.11 release (or really any NetBox release) that would involve modifying table or sequence permissions.

**Reply by candlerb** [reply]
Step (1) is the investigation.

Looking at #6233, it seems that if your database is in a state where some tables are owned by the wrong role, that it's causing a problem with the "standardize_models" migrations.  Perhaps changing the id columns to `models.BigAutoField` has to touch or recreate the sequences for the id columns.

**Reply by jeremystretch** [reply]
> if your database is in a state where some tables are owned by the wrong role

That's my point. NetBox only ever uses a single PostgreSQL role. If ownership of tables or sequences was changed, it was done outside of NetBox.

**Reply by candlerb** [reply]
Absolutely: nobody said it's Netbox's "fault", they just want to know how to diagnose and fix.

In #6233, it looks that Netbox was installed initially with configuration.py connecting as the "postgres" user, and later changed to the "netboxdbuser" user, so tables created before this change have the wrong owner.  I suspect something similar has happened here.

**Reply by tdrevvatne** [reply]
Right - owner is probably the issue here.

netbox=# \dt ipam_vlan
          List of relations
 Schema |   Name    | Type  | Owner  
--------+-----------+-------+--------
 public | ipam_vlan | table | netbox
(1 row)

netbox=# \ds ipam_vlan_id_seq
                List of relations
 Schema |       Name       |   Type   |  Owner   
--------+------------------+----------+----------
 public | ipam_vlan_id_seq | sequence | postgres
(1 row)

I'll try to fix now - will probably solve mye upgrade-"issues"..

**Reply by tdrevvatne** [reply]
Fixing the owner in DB - then upgrade works as a charm...
Thanks for guiding in right direction and help me out...
